metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,643 @@
+            )
+
+            return report_fields
+
+        tile_list_endpoint: str = self.get_tiles_endpoint(
+            workspace, dashboard_id=dashboard.id
+        )
+        # Hit PowerBi
+        logger.info("Request to URL={}".format(tile_list_endpoint))
+        response = requests.get(
+            tile_list_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+        response.raise_for_status()
+
+        # Iterate through response and create a list of PowerBiAPI.Dashboard
+        tile_dict: List[Any] = response.json()[Constant.VALUE]
+        logger.debug("Tile Dict = {}".format(tile_dict))
+        tiles: List[Tile] = [
+            Tile(
+                id=instance.get("id"),
+                title=instance.get("title"),
+                embedUrl=instance.get("embedUrl"),
+                dataset_id=instance.get("datasetId"),
+                **new_dataset_or_report(instance),
+            )
+            for instance in tile_dict
+            if instance is not None
+        ]
+
+        return tiles
+
+
+class RegularAPIResolver(DataResolverBase):
+    # Regular access endpoints
+    API_ENDPOINTS = {
+        Constant.DASHBOARD_LIST: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/dashboards",
+        Constant.TILE_LIST: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/dashboards/{DASHBOARD_ID}/tiles",
+        Constant.DATASET_GET: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/datasets/{DATASET_ID}",
+        Constant.DATASOURCE_GET: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/datasets/{DATASET_ID}/datasources",
+        Constant.REPORT_GET: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/reports/{REPORT_ID}",
+        Constant.REPORT_LIST: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/reports",
+        Constant.PAGE_BY_REPORT: "{POWERBI_BASE_URL}/{WORKSPACE_ID}/reports/{REPORT_ID}/pages",
+    }
+
+    def get_dataset(
+        self, workspace_id: str, dataset_id: str
+    ) -> Optional[PowerBIDataset]:
+        """
+        Fetch the dataset from PowerBi for the given dataset identifier
+        """
+        if workspace_id is None or dataset_id is None:
+            logger.info("Input values are None")
+            logger.info(f"{Constant.WorkspaceId}={workspace_id}")
+            logger.info(f"{Constant.DatasetId}={dataset_id}")
+            return None
+
+        dataset_get_endpoint: str = RegularAPIResolver.API_ENDPOINTS[
+            Constant.DATASET_GET
+        ]
+        # Replace place holders
+        dataset_get_endpoint = dataset_get_endpoint.format(
+            POWERBI_BASE_URL=DataResolverBase.BASE_URL,
+            WORKSPACE_ID=workspace_id,
+            DATASET_ID=dataset_id,
+        )
+        # Hit PowerBi
+        logger.info(f"Request to dataset URL={dataset_get_endpoint}")
+        response = requests.get(
+            dataset_get_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+        # Check if we got response from PowerBi
+        response.raise_for_status()
```

Comment:
What should we do if this raises an exception?
Should we retry or Fail the whole ingestion or should we continue the ingestion?